lib/styled-message.js

```diff
@@ -1,10 +1,28 @@
 const chalk = require('chalk');
+
+/**
+ * @type {(input: string, columns: number) => string }
+ */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: wrap-ansi includes type definitions since version 8, but it also requires node 12
 const wrap = require('wrap-ansi');
 const stripAnsi = require('strip-ansi');
 const terminalLink = require('terminal-link');
 
+/**
+ * @typedef { import("./types/options").Options } Options
+ * @typedef { import('./types/styled-message').FormattedString } FormattedString
+ * @typedef { import("./types/styled-message").StyledMessage } StyledMessage
+ * @typedef { import('./types/styled-message').StyledMessagePart } StyledMessagePart
+ */
+
 // LOG
 
+/**
+ * @param {Options} options
+ * @param {StyledMessage} message
+ * @param {boolean=} clearFixLine
```

Comment:
This is the only question/fix I have, otherwise this looks good :ok_hand:

Reply:
The `=` marks it as an optional parameter.  
I think in one call site it was not passed, so I marked it as optional.

And tsc did not allow me to write `{bool|undefined}`.  
Apparently, there are [these options](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns):

```js
// Parameters may be declared in a variety of syntactic forms
/**
 * @param {string}  p1 - A string param.
 * @param {string=} p2 - An optional param (Google Closure syntax)
 * @param {string} [p3] - Another optional param (JSDoc syntax).
 * @param {string} [p4="test"] - An optional param with a default value
 * @returns {string} This is the result
 */
```

...Or we can change the usage and don't allow undefined.

Reply:
Let's use `@param {string} [p3]` as the syntax, since we're going with JSDoc syntax so far, and it's the one I've been using in other places as well.
I'd be fine with changing the usages as well, whatever you prefer :shrug:

Reply:
I now made the paramter `clearFixLine` a required one with commit 409ad3b